docx/text/paragraph.py

```diff
@@ -616,13 +619,36 @@ def image_parts(self):
         """
         Return all image parts related to this paragraph.
         """
+        doc = self.part.document
         drawings = []
+        parts = []
+
         for r in self.runs:
             if r._element.drawing_lst:
                 drawings.extend(r._element.drawing_lst)
+
         blips = [drawing.xpath(".//*[local-name() = 'blip']")[0]
                  for drawing in drawings]
-        rIds = [b.embed for b in blips]
-        doc = self.part.document
-        parts = [doc.part.related_parts[rId] for rId in rIds]
+
+
+        for b in blips:
+            if b.link:
+                rel = doc.part.rels[b.link]
+                target_ref = pathlib.Path(rel.target_ref)
+                doc_path = pathlib.Path(doc.part.package.path)
+                os.chdir(doc_path.parents[0])
+                rel_path = ''
+
+                for path_part in reversed(target_ref.parts):
+                    rel_path = pathlib.Path(path_part, rel_path)
```

Comment:
If I understand right this code handle inability of Word to store relative path for linked files?

Reply:
yes, exactly. since filepaths are saved under file protocol which doesn't allow relative paths, my guess is that they've made it the way it is for compatibility purposes. more info is available here: https://github.com/openlawlibrary/platform/issues/2097#issuecomment-633681669